.github/workflows/allthekeeps-testnet.yml

```diff
@@ -0,0 +1,103 @@
+    branches:
+     - allthekeeps-on-ropsten
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment for workflow execution'
+        required: false
+        default: 'dev'
+      upstream_builds:
+        description: 'Upstream builds'
+        required: false
+      upstream_ref:
+        description: 'Git reference to checkout (e.g. branch name)'
+        required: false
+        default: 'main'
+
+jobs:
+  dapp-build-and-publish:
+    if: github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: allthekeeps-on-ropsten
+
+      - name: Load environment variables
+        uses: keep-network/ci/actions/load-env-variables@v1
+        with:
+          environment: 'ropsten'
+      
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12.x"
+
+      - name: Run install repositories
+        run: ./install-repositories.sh
+      
+      - name: Get upstream packages versions
+        if: github.event_name == 'workflow_dispatch'
+        uses: keep-network/ci/actions/upstream-builds-query@v1
+        id: upstream-builds-query
+        with:
+          upstream-builds: ${{ github.event.inputs.upstream_builds }}
+          query: |
+            keep-core-contracts-version = github.com/keep-network/keep-core/solidity#version
+            keep-ecdsa-contracts-version = github.com/keep-network/keep-ecdsa/solidity#version
+            tbtc-contracts-version = github.com/keep-network/tbtc/solidity#version
+
+      - name: Deploy subgraph to thegraph.com
+        if: github.event_name != 'workflow_dispatch'
+        working-directory: ./keep-subgraph
+        run: |
+          CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY=${{secrets.CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY}} \
+          ETH_RPC_URL=${{secrets.ETH_RPC_URL}} \
+          SUBGRAPH_DEPLOY_KEY=${{secrets.SUBGRAPH_DEPLOY_KEY}} \
+          SUBGRAPH_SLUG=${{secrets.SUBGRAPH_SLUG}} \
+            ./deploy-subgraph.sh
+
+      - name: Deploy latest subgraph to thegraph.com
+        if: github.event_name == 'workflow_dispatch'
+        working-directory: ./keep-subgraph
+        run: | 
+          CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY=${{ secrets.CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY }} \
+          ETH_RPC_URL=${{ secrets.ETH_RPC_URL }} \ 
+          SUBGRAPH_DEPLOY_KEY=${{ secrets.SUBGRAPH_DEPLOY_KEY }} \ 
+          SUBGRAPH_SLUG=${{ secrets.SUBGRAPH_SLUG }} \ 
+          KEEP_CORE_VERSION=${{ steps.upstream-builds-query.outputs.keep-core-contracts-version }} \
+          KEEP_ECDSA_VERSION=${{ steps.upstream-builds-query.outputs.keep-ecdsa-contracts-version }} \
+          KEEP_TBTC_VERSION=${{ steps.upstream-builds-query.outputs.tbtc-contracts-version }} \
+            ./deploy-subgraph.sh
+
+      - name: YARN build
+        run: ./install-dapp.sh
+
+      - name: Deploy to GCP bucket
+        if: github.event_name == 'workflow_dispatch'
```

Comment:
We deploy to GCP only on `workflow_dispatch`. Maybe we should do the same for `Deploy subgraph to thegraph.com`?

Reply:
Practically, I think that in most cases `Deploy subgraph to thegraph.com` won't be triggered, cause atk is a forked repo and there were no plans (as far as I know) to work on the code. I've added `Deploy subgraph to thegraph.com` mostly to check that the dots (deploying to The Graph, building everything except publishing the dApp) are connecting when we run into a situation when we want to test some minor changes and I don't think it's necessary to publish on GCP when testing manually. In most cases `workflow_dispatch ` is an event that will trigger everything.

Reply:
ok, done 3f59695623de142141a74bc77c7ebe98db1b4f79